pdm/models/candidates.py

```diff
@@ -409,10 +410,12 @@ def _populate_source_dir(self, ireq: pip_shims.InstallRequirement) -> None:
         elif self.req.editable:
             if self.environment.packages_path:
                 src_dir = self.environment.packages_path / "src"
-            elif os.getenv("VIRTUAL_ENV"):
-                src_dir = Path(os.environ["VIRTUAL_ENV"]) / "src"
             else:
-                src_dir = Path("src")
+                virtual_env_prefix = get_env_prefix()
```

Comment:
I am thinking it shouldn't be testing the env var only, but it should check `is_venv_python(self.environment.interpreter.executable)`

Reply:
I think you're right. In fact, I don't think we should really be checking `venv` outside of `Project`, because once we have chosen the interpreter, we want to use the same sysconfig information everywhere.

Reply:
This is to keep the same behavior of `pip`, so that users know where to find the source when using an editable install.

1. venv pyhon: `<venv_prefix>/src`
2. non-venv python: `$(pwd)/src`
3. (PEP 582 specific): `__pypackages__/$major.$minor/src`

Reply:
Will reply below in comments :)